### PR TITLE
*: Handle invalid schema like failed commit when finishing txn

### PR DESCRIPTION
**Pull request by zimulala**
If the function of `checkSchemaValid` returns an error and the error isn't retryable, it will do rollback.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session.go [hunk 1]
```diff
@@ -240,24 +240,19 @@ func (s *session) finishTxn(rollback bool) error {
 		}
 	}
 
-	if err := s.checkSchemaValid(); err != nil {
-		if !s.sessionVars.RetryInfo.Retrying && s.isRetryableError(err) {
-			err = s.Retry()
-		} else {
+	err := s.checkSchemaValid()
+	if err != nil {
+		if !s.isRetryableError(err) {
```
Comment (shenli) [inline session.go]: Should we return in this branch?

Comment (coocood) [inline session.go]: No, there are clean up works to do.

Comment (coocood) [inline session.go]: But resetHistory will not be called, it will return at line 261.

Comment (shenli) [inline session.go]: But it will return at line 261 with nothing to do about clean up works.

**Comment by zimulala** [comment]
PTAL @shenli

**Pull request closed without merging**